### PR TITLE
[CGData][llvm-cgdata] Support for stable function map

### PR DESCRIPTION
**Pull request by kyulee-com**
This introduces a new cgdata format for stable function maps. The raw data is embedded in the __llvm_merge section during compile time. This data can be read and merged using the llvm-cgdata tool, into an indexed cgdata file. Consequently, the tool is now capable of handling either outlined hash trees, stable function maps, or both, as they are orthogonal.

Depends on #112662.
This is a patch for https://discourse.llvm.org/t/rfc-global-function-merging/82608.

### CODE REVIEW

**Comment by kyulee-com** [comment]
cc. @nocchijiang

File: llvm/lib/CGData/CodeGenDataReader.cpp [hunk 1]
```diff
@@ -32,10 +32,40 @@ setupMemoryBuffer(const Twine &Filename, vfs::FileSystem &FS) {
 
 Error CodeGenDataReader::mergeFromObjectFile(
     const object::ObjectFile *Obj, OutlinedHashTreeRecord &GlobalOutlineRecord,
+    StableFunctionMapRecord &GlobalFunctionMapRecord,
     stable_hash *CombinedHash) {
   Triple TT = Obj->makeTriple();
   auto CGOutLineName =
       getCodeGenDataSectionName(CG_outline, TT.getObjectFormat(), false);
```
Comment (ellishg) [inline llvm/lib/CGData/CodeGenDataReader.cpp:39]: I think this should be spelled `GCOutlineName` since it's "outline", not "out line".

**Comment by kyulee-com** [comment]
Are there any additional comments? I believe this change is relatively straightforward.

**Review by thevinster — APPROVED** [review]
lgtm

**Pull request merged**